### DISCUSSION TITLE
How to check if two enumerations have the same values?

### DISCUSSION THREAD

**Discussion by PatryxCShark** [shouldly/shouldly#785] [Q&A] [answered]
Hi,
I want to check if two collections with GUID have all the same items, but the order is not important. for me
`IEnumerable<Guid> list1;
IEnumerable<Guid> list2;`

I can use:
;`list1.ShouldBe(list2);`
but here the order is important.

Is there any possibility to do it nicely in Shouldly?

**Comment by jamesmcroft** [comment]
I ended up creating a custom extension method for this scenario called `ShouldBeEquivalentTo` which basically performs the check without a care for the order. 

I'm not sure if there is a way of doing this in the Shoudly framework but I would always be interested in knowing if there is a way of achieving this. 

Alternatively, very open to providing a contribution for this if not 😄

**Comment by SimonCropp** [comment]
happy to accept a pr. thanks

**Comment by jamesmcroft** [comment]
So I've looked into this and somehow missed it. Apparently, there is a `ShouldBeEquivalentTo` extension. However, it doesn't support out of order items.

**Comment by jnm2** [comment, marked as answer]
@PatryxCShark Yes, like this:

```cs
list1.ShouldBe(list2, ignoreOrder: true);
```

**Reply by PatryxCShark** [reply]
Thanks!
